src/hls-adapter.js

```diff
@@ -182,6 +192,20 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     return this._loadPromise;
   }
 
+  /**
+   * Loaded metadata event handler.
+   * @param {Function} resolve - The resolve promise function.
+   * @private
+   * @returns {void}
+   */
+  _onLoadedMetadata(resolve: Function): void {
+    if (this._onLoadedMetadataCallback) {
```

Comment:
You have same exact code in lines 202:205 and lines 643:646.
Extract to one method call from both places

Reply:
done